tools/clang/unittests/HLSL/ValidationTest.cpp

```diff
@@ -4207,6 +4211,127 @@ TEST_F(ValidationTest, ValidateWithHash) {
+#ifdef _WIN32
+  VERIFY_IS_TRUE(SetEnvironmentVariableW(varName.c_str(), varValue.c_str()));
+  // also update the CRT environment
+  std::string varNameStr;
+  std::string varValueStr;
+  Unicode::WideToUTF8String(varName.c_str(), &varNameStr);
+  Unicode::WideToUTF8String(varValue.c_str(), &varValueStr);
+  _putenv_s(varNameStr.c_str(), varValueStr.c_str());
+#else
+  std::string name_utf8 = wstring_to_utf8(varName);
+  std::string value_utf8 = wstring_to_utf8(varValue);
+  setenv(name_utf8.c_str(), value_utf8.c_str(), 1);
+#endif
+}
+
+void ClearEnvVarW(const std::wstring &varName) {
+  std::string varNameStr;
+  Unicode::WideToUTF8String(varName.c_str(), &varNameStr);
+#ifdef _WIN32
+  SetEnvironmentVariableW(varName.c_str(), nullptr);
+  _putenv_s(varNameStr.c_str(), "");
+#else
+  unsetenv(varNameStr.c_str());
+#endif
+}
+
+// For now, 3 things are tested:
+// 1. The environment variable is not set. GetDxilDllPath() is empty and
+// DxilDllFailedToLoad() returns false
+// 2. Given a bogus path in the environment variable, GetDxilDllPath()
+// retrieves the path but fails to load it as a dll, and returns true
+// for DxilDllFailedToLoad()
+// 3. CLSID_DxcCompiler, CLSID_DxcLinker, CLSID_DxcValidator
+// may be created through DxcDllExtValidationSupport.
+// This is all to simply test that the new class, DxcDllExtValidationSupport,
+// works as intended.
+
+TEST_F(ValidationTest, UnitTestExtValidationSupport) {
+  DxcDllExtValidationSupport m_dllExtSupport1;
+  DxcDllExtValidationSupport m_dllExtSupport2;
+
+  // capture any existing value in the environment variable,
+  // so that it can be restored after the test
+  std::wstring oldEnvVal = GetEnvVarW(L"DXC_DXIL_DLL_PATH");
+
+  // 1. with no env var set, test GetDxilDllPath() and DxilDllFailedToLoad()
+
+  // make sure the variable is cleared, in case other tests may have set it
+  SetEnvVarW(L"DXC_DXIL_DLL_PATH", L"");
+
+  // empty initialization should succeed
+  VERIFY_SUCCEEDED(m_dllExtSupport1.Initialize());
+
+  VERIFY_IS_FALSE(m_dllExtSupport1.DxilDllFailedToLoad());
+  VERIFY_ARE_EQUAL(m_dllExtSupport1.GetDxilDllPath(), "");
+
+  // 2. Test with a bogus path in the environment variable
+  SetEnvVarW(L"DXC_DXIL_DLL_PATH", L"bogus");
+
+  if (!m_dllExtSupport2.IsEnabled()) {
+    VERIFY_SUCCEEDED(m_dllExtSupport2.Initialize());
+  }
+
+  // validate that m_dllExtSupport2 was able to capture the environment
+  // variable's value, and that loading the bogus path was unsuccessful
+  std::string extPath("bogus");
+  VERIFY_ARE_EQUAL(m_dllExtSupport2.GetDxilDllPath(), extPath);
+  VERIFY_IS_TRUE(m_dllExtSupport2.DxilDllFailedToLoad());
+
+  // 3. Test production of class IDs CLSID_DxcCompiler, CLSID_DxcLinker,
+  // and CLSID_DxcValidator through DxcDllExtValidationSupport.
+  CComPtr<IDxcCompiler> pCompiler;
+  CComPtr<IDxcLinker> pLinker;
+  CComPtr<IDxcValidator> pValidator;
+  CComPtr<IDxcOperationResult> pResult;
```

Comment:
`pResult` is unused.
Plus, naming: even though these follow a rough convention used elsewhere, we should follow the coding standards for new code.
```suggestion
  CComPtr<IDxcCompiler> Compiler;
  CComPtr<IDxcLinker> Linker;
  CComPtr<IDxcValidator> Validator;
```